sources/controllers/node_controller.rb

```diff
@@ -11,28 +9,35 @@ class NodeController < Sinatra::Base
   end
 
   # html view, sends stuff to /node as POST
-  get '/new' do
-    @node = Node.new
-    haml :node
-  end
-
-  post '/' do
-    Node.new(params[:node])
-  end
+  get '/new' do             #
+    @node = Node.new        # dafuq is this
+    haml :node              #
+  end                       #
 
-  delete '/:id' do
-    Node.delete(params[:id])
-  end
+  post '/' do               #
+    Node.new(params[:node]) # dafuq is that
+  end                       #
 
-  put '/:id' do
+  before %r{\A/(?<id>\d+)/?.*} do
     @node = Node.find(params[:id])
-    # how do we update all provided params?
-    # remove id, than pass hash to node.update()?
-    redirect to("/nodes/#{node.id}")
   end
 
-  get '/:id' do
-    @node = Node.find(params[:id])
-    haml :node
+  namespace '/:id' do
+
+    delete do
+      @node.delete
+    end
+
+    put do
+      # how do we update all provided params?
+        # there must be a way
+      # remove id, than pass hash to node.update()?
+        # doesn't sound THAT great, .assign_attributes might be a good choice?
```

Comment:
so it could be:
```ruby
put do
  @node.assign_attributes(params[:node])
```

?
Could you add this to the PR and remove the "# dafuq is this"  than :+1: for merging